core/cache/unit.go

```diff
@@ -181,6 +181,11 @@ func (u *Unit) WatchConfigSettings() (*CharmConfigWatcher, error) {
 	return w, errors.Trace(err)
 }
 
+// GetDetails returns the current unit details as a change type.
+func (u *Unit) GetDetails() UnitChange {
+	return u.details.copy()
```

Comment:
I'm curious. Is this completely necessary? This may lead to a memory leak.

Reply:
The unit change isn't a ptr so we're copying by value. @achilleasa had a moan about this as well.